core/src/test/java/org/apache/iceberg/rest/responses/TestPlanTableScanResponseParser.java

```diff
@@ -0,0 +1,244 @@
+
+    String invalidJson = "{\"plan-status\":\"submitted\"}";
+    assertThatThrownBy(() -> PlanTableScanResponseParser.fromJson(invalidJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: plan id should be defined when status is 'submitted'");
+  }
+
+  @Test
+  public void roundTripSerdeWithInvalidPlanStatusCancelled() {
+    PlanStatus planStatus = PlanStatus.fromName("cancelled");
+    assertThatThrownBy(() -> PlanTableScanResponse.builder().withPlanStatus(planStatus).build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: 'cancelled' is not a valid status for planTableScan");
+
+    String invalidJson = "{\"plan-status\":\"cancelled\"}";
+    assertThatThrownBy(() -> PlanTableScanResponseParser.fromJson(invalidJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: 'cancelled' is not a valid status for planTableScan");
+  }
+
+  @Test
+  public void roundTripSerdeWithInvalidPlanStatusSubmittedWithTasksPresent() {
+    PlanStatus planStatus = PlanStatus.fromName("submitted");
+    assertThatThrownBy(
+            () ->
+                PlanTableScanResponse.builder()
+                    .withPlanStatus(planStatus)
+                    .withPlanId("somePlanId")
+                    .withPlanTasks(List.of("task1", "task2"))
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: tasks can only be returned in a 'completed' status");
+
+    String invalidJson =
+        "{\"plan-status\":\"submitted\","
+            + "\"plan-id\":\"somePlanId\","
+            + "\"plan-tasks\":[\"task1\",\"task2\"]}";
+
+    assertThatThrownBy(() -> PlanTableScanResponseParser.fromJson(invalidJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: tasks can only be returned in a 'completed' status");
+  }
+
+  @Test
+  public void roundTripSerdeWithInvalidPlanIdWithIncorrectStatus() {
+    PlanStatus planStatus = PlanStatus.fromName("failed");
+    assertThatThrownBy(
+            () ->
+                PlanTableScanResponse.builder()
+                    .withPlanStatus(planStatus)
+                    .withPlanId("somePlanId")
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: plan id can only be returned in a 'submitted' status");
+
+    String invalidJson = "{\"plan-status\":\"failed\"," + "\"plan-id\":\"somePlanId\"}";
+
+    assertThatThrownBy(() -> PlanTableScanResponseParser.fromJson(invalidJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: plan id can only be returned in a 'submitted' status");
+  }
+
+  @Test
+  public void roundTripSerdeWithInvalidPlanStatusSubmittedWithDeleteFilesNoFileScanTasksPresent() {
```

Comment:
I don't think any of the failure case tests should have the prefix "round trip", nothing is really roundtripping here

Reply:
Will fix this